myenumerables.rb

```diff
@@ -0,0 +1,158 @@
+        return false unless idx == query
+      end
+    elsif query.class == Regexp
+      my_each do |idx|
+        return false unless idx.match(query)
+      end
+    elsif block_given?
+      my_each do |idx|
+        return false unless yield(idx)
+      end
+    elsif query.nil?
+      my_each do |idx|
+        return false if idx === false
+      end
+    else
+      my_each do |idx|
+        return false unless !idx.nil?
+      end
+    end
+    true
+  end
+
+  def my_any?(*query)
+    if block_given?
+      length.times { |item| return true unless yield(item) }
+    elsif query.class == Regexp
+      length.times { |item| return true unless item == query }
+    elsif query.class == Class
+      length.times { |item| return true unless item.class == query }
+    elsif query.class == Numeric or query.class == String
+      length.times { |item| return true unless item == query }
+    elsif query.class == Proc
+      length.times { |item| return true unless yield(item) }
+    else
+      length.times { |item| return true unless item }
+    end
+    false
+  end
+
+  def my_none?(*query)
+    if block_given?
+      length.times { |item| return false unless yield(item) }
+    elsif query.class == Regexp
+      length.times { |item| return false unless item == query }
+    elsif query.class == Class
+      length.times { |item| return false unless item.class == query }
+    elsif query.class == Numeric or query.class == String
+      length.times { |item| return false unless item == query }
+    else
+      length.times { |item| return false unless item }
+    end
+    true
+  end
+
+  def my_count(query = nil)
+    array = self
+    count = 0
+    if block_given?
+      array.length.times { |item| count += 1 if yield(array[item]) }
+    elsif !query.nil?
+      array.length.times { |item| count += 1 if array[item] == query }
+    elsif query.nil?
+      count = array.length
+    end
+    count
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    array = []
+    if block_given?
+      to_a.my_each do |item|
+        array.push(yield(item))
+      end
+    end
+    array
+  end
+
+  def my_inject(query = nil, query2 = nil)
+    array = to_a
+    block_return = ''.to_i
+    if block_given? and query.nil?
+      block_return = array.shift
+      for item in array do
+        block_return = yield(block_return, item)
+      end
+      block_return
+    elsif block_given? and query2.nil? and query.class == Integer
+      array.push(query)
+      block_return = array.shift
+      for item in array do
+        block_return = yield(block_return, item)
+      end
+      block_return
+    elsif query2.class == Symbol
+      sum = query
+      array.my_each do |item|
+        sum = sum.public_send(query2, item)
+      end
+      sum
+    elsif query.class == Symbol
+      array.reduce(0) { |sum, num| sum << num.public_send(query, num + 1) }
+      sum
+    else
+      array.push(query)
+      array
+    end
+  end
```

Comment:
```rb
2.6.3 :074 > array
 => [3, 8, 5, 6, 1, 5, 8, 3, 7, 7, 6, 0, 3, 6, 7, 7, 2, 2, 1, 1, 4, 3, 5, 1, 4, 5, 2, 4, 8, 7, 2, 7, 8, 2, 2, 7, 4, 0, 6, 5, 7, 7, 6, 4, 1, 8, 5, 5, 7, 0, 3, 3, 1, 0, 1, 5, 5, 7, 3, 8, 5, 2, 1, 5, 3, 8, 6, 3, 5, 5, 6, 4, 0, 7, 5, 7, 7, 7, 7, 8, 1, 6, 8, 3, 1, 6, 5, 6, 5, 7, 4, 1, 8, 8, 6, 1, 1, 0, 8, 5] 
2.6.3 :075 > operation = proc { |sum, n| sum + n }
 => #<Proc:0x00000000021413b8@(irb):75> 
2.6.3 :076 > array.inject(&operation)
 => 451 
2.6.3 :077 > array.my_inject(&operation)
 => 451 
2.6.3 :078 > array.inject(&operation)
 => 448 
2.6.3 :079 > array.my_inject(&operation)
 => 448 
2.6.3 :080 > array.my_inject(&operation)
 => 440 
2.6.3 :081 > array.inject(&operation)
 => 435 
2.6.3 :082 > 
```
There is something wrong !

Reply:
```rb
2.6.3 :084 > array_clone = array.clone
 => [6, 1, 5, 8, 3, 7, 7, 6, 0, 3, 6, 7, 7, 2, 2, 1, 1, 4, 3, 5, 1, 4, 5, 2, 4, 8, 7, 2, 7, 8, 2, 2, 7, 4, 0, 6, 5, 7, 7, 6, 4, 1, 8, 5, 5, 7, 0, 3, 3, 1, 0, 1, 5, 5, 7, 3, 8, 5, 2, 1, 5, 3, 8, 6, 3, 5, 5, 6, 4, 0, 7, 5, 7, 7, 7, 7, 8, 1, 6, 8, 3, 1, 6, 5, 6, 5, 7, 4, 1, 8, 8, 6, 1, 1, 0, 8, 5] 
2.6.3 :085 >       array.my_inject { |num| num + 1}
 => 102 
2.6.3 :086 > array == array_clone
 => false 
2.6.3 :087 > array === array_clone
 => false 
2.6.3 :088 >
```
They should be equal

Reply:
```ruby
2.6.3 :090 > array_clone = array.clone
 => [1, 5, 8, 3, 7, 7, 6, 0, 3, 6, 7, 7, 2, 2, 1, 1, 4, 3, 5, 1, 4, 5, 2, 4, 8, 7, 2, 7, 8, 2, 2, 7, 4, 0, 6, 5, 7, 7, 6, 4, 1, 8, 5, 5, 7, 0, 3, 3, 1, 0, 1, 5, 5, 7, 3, 8, 5, 2, 1, 5, 3, 8, 6, 3, 5, 5, 6, 4, 0, 7, 5, 7, 7, 7, 7, 8, 1, 6, 8, 3, 1, 6, 5, 6, 5, 7, 4, 1, 8, 8, 6, 1, 1, 0, 8, 5] 
2.6.3 :091 > array.inject(:+)
 => 429 
2.6.3 :092 > array_clone.my_inject(:+)
 => nil 
2.6.3 :093 > 
```

Reply:
That error https://github.com/rootDEV2990/ruby_enumerables_rewrites/pull/1#discussion_r429651463 still persist

Reply:
I rewrote yesterday does new one not work?


def my_each
    index = 0
    array = to_a
    if block_given?
      for item in 0..array.length - 1
        yield(self[item])
      end
    else
      while index < array.size
        return to_enum(:my_each) unless block_given?

        yield self[index]
        index += 1
        return self
      end
    end
  end

Reply:
Thanks for the help, I re coded some stuff if you can help me see it now im not sure im testing this right. Thanks.